src/sap/fhir/model/r4/FHIRFilterOperatorUtils.js

```diff
@@ -226,18 +226,39 @@ sap.ui.define([
 	 */
 	FHIRFilterOperatorUtils.getFilterValueForComplexFilter = function (sFilterValue, vValue) {
 		var sValue;
-		// special handling for string parameter as per fhir
-		// given eq "peter"
-		if (sFilterValue && sFilterValue === FHIRFilterType.string) {
-			sValue = "\"" + vValue + "\"";
-		} else if (typeof vValue === "string" && isNaN(new Date(vValue).getTime())) {
-			// incase of date as string it shouldnt be encoded
+		if (this.canBeEncoded(sFilterValue, vValue)) {
 			sValue = "\"" + vValue + "\"";
 		} else {
 			sValue = vValue;
 		}
 		return sValue;
 	};
 
+	/**
+	 * Determines if the value should be encoded or not
+	 *
+	 * @param {string} sFilterValue The value type of a filter object
+	 * @param {any} vValue The value of a filter object
+	 * @returns {boolean} true if the value needs to be encoded
+	 * @private
+	 * @since 2.2.7
+	 */
+	FHIRFilterOperatorUtils.canBeEncoded = function (sFilterValue, vValue) {
+		var sRegex = "[ \r\n\t\S]+";
+		return (sFilterValue && sFilterValue === FHIRFilterType.string) || (typeof vValue === "string" && (vValue.match(sRegex) != null || this.isValidDate(vValue)));
+	};
+
+	/**
+	 * Determines if the value is valid date object
```

Comment:
Determines if the given <code>vValue</code> can be parsed to a valid date object

Reply:
done